Izanami client 1.4.0 fails to connect to cloud hosted server

We have the izanami client running in our service. Our service is running Play 2.7 on Scala 2.12.8.
We included some new dependencies in hopes of squashing some dependency conflict warnings when booting up the service, namely: 

  "com.typesafe.akka"           %% "akka-parsing"                   % "10.1.7",
  "com.typesafe.akka"           %% "akka-http"                         % "10.1.7",
  "com.typesafe.akka"           %% "akka-http-core"                 % "10.1.7"

Upon further inspection, I saw that izanami uses "akka-http" at version 10.1.3, so I also tried changing our dependencies accordingly and the issue still stands.

Basically, having the izanami server (1.4.0) in a dockerized container locally running on localhost:8081 works as expected, but the izanami client fails to communicate to a cloud hosted instance of the izanami client.

Here are parts of our logs:
First, it starts off with some warnings:

``` 
[WARN] [03/08/2019 09:12:48.006] [IzanamiSystem-akka.actor.default-dispatcher-3] [IzanamiSystem/Pool(shared->*cloud hosted server location*:0)] Connection attempt failed. Backing off new connection attempts for at least 100 milliseconds.

[WARN] [03/08/2019 09:12:48.166] [IzanamiSystem-akka.actor.default-dispatcher-3] [IzanamiSystem/Pool(shared->*cloud hosted server location*:0)] Connection attempt failed. Backing off new connection attempts for at least 200 milliseconds. 
```

which then escalate to errors:

```
[ERROR] [03/08/2019 09:15:05.539] [IzanamiSystem-akka.actor.default-dispatcher-7] [akka.actor.ActorSystemImpl(IzanamiSystem)] Error during call
akka.stream.BufferOverflowException: Exceeded configured max-open-requests value of [32]. 
This means that the request queue of this pool (HostConnectionPoolSetup(*cloud hosted server location*,0,ConnectionPoolSetup(ConnectionPoolSettings(4,0,5,32,1,100 milliseconds,2 minutes,30 seconds,ClientConnectionSettings(Some(User-Agent: akka-http/10.1.7),10 seconds,1 minute,512,None,WebSocketSettings(<function0>,ping,Duration.Inf,akka.http.impl.settings.WebSocketSettingsImpl$$$Lambda$578/0x000000084049e840@4f134fc2),List(),ParserSettings(2048,16,64,64,8192,64,8388608,8388608,256,1048576,Strict,RFC6265,true,Set(),Full,Error,Map(If-Range -> 0, If-Modified-Since -> 0, If-Unmodified-Since -> 0, default -> 12, Content-MD5 -> 0, Date -> 0, If-Match -> 0, If-None-Match -> 0, User-Agent -> 32),false,true,akka.util.ConstantFun$$$Lambda$568/0x0000000840470840@630f9c9,akka.util.ConstantFun$$$Lambda$568/0x0000000840470840@630f9c9,akka.util.ConstantFun$$$Lambda$569/0x0000000840498040@1ab1a2a1),None,TCPTransport),New,1 second),akka.http.scaladsl.HttpConnectionContext$@4f09936a,akka.event.MarkerLoggingAdapter@840e350))) has completely filled up because the pool currently does not process requests fast enough to handle the incoming request load. Please retry the request later. See http://doc.akka.io/docs/akka-http/current/scala/http/client-side/pool-overflow.html for more information.
at akka.http.impl.engine.client.PoolInterfaceActor.<init>(PoolInterfaceActor.scala:88)
at akka.http.impl.engine.client.PoolInterfaceActor$.$anonfun$props$1(PoolInterfaceActor.scala:34)
at akka.actor.TypedCreatorFunctionConsumer.produce(IndirectActorProducer.scala:87)...

[ERROR] [03/08/2019 09:15:05.540] [IzanamiSystem-akka.actor.default-dispatcher-7] [SmartCacheFeatureClient(akka://IzanamiSystem)] Error fetching features with pattern myfeature:myfeature:*

[ERROR] [03/08/2019 09:15:05.541] [IzanamiSystem-akka.actor.default-dispatcher-10] [SmartCacheStrategyHandler(akka://IzanamiSystem)] Error refreshing cache 
```

We tried using the Fetch strategy at first and then tried the CacheWithPollingStrategy, with the same results.

Another question would be: If online connection fails, how come it doesn't just simply fall back to the fallback configuration?

--- a/izanami-clients/jvm/build.sbt
+++ b/izanami-clients/jvm/build.sbt
@@ -1,13 +1,14 @@
 import sbt.Keys.{organization, scalacOptions}
 import sbtrelease.ReleaseStateTransformations._
 
-val akkaVersion = "2.5.17"
-
 val disabledPlugins = if (sys.env.get("TRAVIS_TAG").filterNot(_.isEmpty).isDefined) {
   Seq(RevolverPlugin)
 } else {
   Seq(RevolverPlugin, BintrayPlugin)
 }
+
+val akkaVersion     = "2.5.19"
+val akkaHttpVersion = "10.1.7"
 
 lazy val jvm = (project in file("."))
   .disablePlugins(disabledPlugins: _*)
@@ -18,12 +19,12 @@ lazy val jvm = (project in file("."))
     libraryDependencies ++= Seq(
       "com.typesafe.akka"          %% "akka-stream"             % akkaVersion,
       "com.typesafe.akka"          %% "akka-slf4j"              % akkaVersion,
-      "com.typesafe.akka"          %% "akka-http"               % "10.1.3",
+      "com.typesafe.akka"          %% "akka-http"               % akkaHttpVersion,
       "com.lightbend.akka"         %% "akka-stream-alpakka-sse" % "0.19",
       "io.vavr"                    % "vavr"                     % "0.9.2",
       "org.reactivecouchbase.json" % "json-lib"                 % "1.0.0",
       "com.google.guava"           % "guava"                    % "25.1-jre",
-      "com.typesafe.play"          %% "play-json"               % "2.6.9",
+      "com.typesafe.play"          %% "play-json"               % "2.7.0",
       "com.chuusai"                %% "shapeless"               % "2.3.3",
       "com.adelegue"               %% "playjson-extended"       % "0.0.4",
       "junit"                      % "junit"                    % "4.12" % Test,

--- a/izanami-clients/jvm/src/main/resources/reference.conf
+++ b/izanami-clients/jvm/src/main/resources/reference.conf
@@ -18,7 +18,7 @@ akka.http {
   host-connection-pool {
     # The maximum number of parallel connections that a connection pool to a
     # single host endpoint is allowed to establish. Must be greater than zero.
-    max-connections = 4
+    max-connections = 128
 
     # The minimum number of parallel connections that a pool should keep alive ("hot").
     # If the number of connections is falling below the given threshold, new ones are being spawned.
@@ -38,7 +38,7 @@ akka.http {
     # Note that with N concurrent materializations the max number of open request in the pool
     # will never exceed N * max-connections * pipelining-limit.
     # Must be a power of 2 and > 0!
-    max-open-requests = 32
+    max-open-requests = 128
 
     # The maximum number of requests that are dispatched to the target host in
     # batch-mode across a single connection (HTTP pipelining).

--- a/izanami-clients/jvm/src/main/scala/izanami/commons/client.scala
+++ b/izanami-clients/jvm/src/main/scala/izanami/commons/client.scala
@@ -66,25 +66,9 @@ private[izanami] class HttpClient(system: ActorSystem, config: ClientConfig) {
 
   private val uri = Uri(config.host)
 
-  private val http: Flow[(HttpRequest, Unit), (Try[HttpResponse], Unit), HostConnectionPool] = uri.scheme match {
-    case "https" =>
-      logger.info(s"Creating https connection pool for {}:{}", uri.authority.host.address(), uri.authority.port)
-      Http().cachedHostConnectionPoolHttps(uri.authority.host.address(), uri.authority.port)
-    case _ =>
-      logger.info(s"Creating http connection pool for {}:{}", uri.authority.host.address(), uri.authority.port)
-      Http().cachedHostConnectionPool(uri.authority.host.address(), uri.authority.port)
-  }
+  private val http = Http()
 
-  private def singleRequest(r: HttpRequest): Future[HttpResponse] =
-    Source
-      .single(r -> ())
-      .via(http)
-      .map(_._1)
-      .flatMapConcat {
-        case Success(s) => Source.single(s)
-        case Failure(e) => Source.failed(e)
-      }
-      .runWith(Sink.head)
+  private def singleRequest(r: HttpRequest): Future[HttpResponse] = http.singleRequest(r)
 
   private val headers = (for {
     clientId     <- config.clientId

